src/app/code/community/Alma/Installments/Helper/Eligibility.php

```diff
@@ -27,143 +27,144 @@
-        $nonEligibilityMessage = $this->config->getNonEligibilityMessage();
-        $excludedProductsMessage = $this->config->getExcludedProductsMessage();
+    public function getEligibleFeePlans()
+    {
+       if($this->eligibleFeePlansAreLoaded){
+           return $this->currentEligibleFeePlans;
+       }
+        $feePlansEligibilities = $this->getAlmaFeePlansEligibility();
+        $eligibleFeePlans = $this->selectEligibleFeePlans($feePlansEligibilities);
+        $this->saveEligibleFeePlans($eligibleFeePlans);
+        return $eligibleFeePlans;
+    }
 
-        if (!$this->alma) {
-            $this->eligible = false;
-            return false;
+    /**
+     * @return array|void
+     * @throws Mage_Core_Exception
+     */
+    private function getAlmaFeePlansEligibility()
+    {
+        if(!$this->checkEligibilityPrerequisite()){
+            return [];
         }
-
-        if (!$this->checkItemsTypes()) {
-            $this->eligible = false;
-            $this->message = $nonEligibilityMessage . '<br>' . $excludedProductsMessage;
-            return false;
+        $quote = Mage::helper(self::MAGE_CART_KEY)->getQuote();
+        $cartTotal = Alma_Installments_Helper_Functions::priceToCents((float)$quote->getGrandTotal());
+        $enabledFeePlansInConfig = $this->feePlansHelper->getEnabledFeePlansConfigFromBackOffice();
+
+        $installmentsQuery = [];
+        foreach ($enabledFeePlansInConfig as $configFeePlan) {
+            if (
+                $cartTotal >= $configFeePlan[Alma_Installments_Helper_FeePlansHelper::MIN_DISPLAY_KEY] &&
+                $cartTotal <= $configFeePlan[Alma_Installments_Helper_FeePlansHelper::MAX_DISPLAY_KEY]
+            ){
+                $installmentsQuery[] = [
+                    'purchase_amount' => $cartTotal,
+                    'installments_count' => $configFeePlan['installments_count'],
+                    'deferred_days' => $configFeePlan['deferred_days'],
+                    'deferred_month' => $configFeePlan['deferred_months'],
+                    'cart_total' => $cartTotal,
+                ];
+            }
+        }
+        if (empty($installmentsQuery)) {
+            return [];
         }
 
-        /** @var Mage_Sales_Model_Quote $quote */
-        $quote = Mage::helper('checkout/cart')->getQuote();
-        if(!$quote) {
-            $this->eligible = false;
-            return false;
+        try {
+            $feePlansEligibilities = $this->alma->payments->eligibility(
+                $this->formatEligibilityPayload($quote, $installmentsQuery),
+                true
+            );
+        } catch (RequestError $e) {
+            $this->logger->error('$e',[$e->getMessage()]);
+            return [];
         }
+        return $feePlansEligibilities;
+    }
 
-        $this->message = $eligibilityMessage;
-        $cartTotal = Alma_Installments_Helper_Functions::priceToCents((float)$quote->getGrandTotal());
+    /**
+     * @param $eligibleFeePlans
```

Comment:
please give type for this `@param`

Reply:
done and remove :           
/** @var Alma\API\Endpoints\Results\Eligibility $feePlansEligibility */